contracts/asset-proxy/test/erc20bridge_proxy.ts

```diff
@@ -0,0 +1,298 @@
+            const tx = transferFromAsync({}, badCaller);
+            return expect(tx).to.revertWith(new AuthorizableRevertErrors.SenderNotAuthorizedError(badCaller));
+        });
+
+        it('fails if asset data is truncated', async () => {
+            const opts = createTransferFromOpts();
+            const truncatedAssetData = hexSlice(encodeAssetData(opts.assetData), 0, -1);
+            const tx = assetProxy.transferFrom.awaitTransactionSuccessAsync(
+                truncatedAssetData,
+                opts.from,
+                opts.to,
+                new BigNumber(opts.amount),
+            );
+            return expect(tx).to.be.rejected();
+        });
+
+        it('fails if bridge returns nothing', async () => {
+            const tx = transferFromAsync({
+                assetData: createAssetData({
+                    bridgeData: createBridgeData({
+                        returnData: '0x',
+                    }),
+                }),
+            });
+            // This will actually revert when the AP tries to decode the return
+            // value.
+            return expect(tx).to.be.rejected();
+        });
+
+        it('fails if bridge returns true', async () => {
+            const tx = transferFromAsync({
+                assetData: createAssetData({
+                    bridgeData: createBridgeData({
+                        returnData: hexLeftPad('0x1'),
+                    }),
+                }),
+            });
+            // This will actually revert when the AP tries to decode the return
+            // value.
+            return expect(tx).to.be.rejected();
+        });
+
+        it('fails if bridge returns 0x1', async () => {
+            const tx = transferFromAsync({
+                assetData: createAssetData({
+                    bridgeData: createBridgeData({
+                        returnData: hexRightPad('0x1'),
+                    }),
+                }),
+            });
+            return expect(tx).to.revertWith('BRIDGE_FAILED');
+        });
+
+        it('fails if bridge is an EOA', async () => {
+            const tx = transferFromAsync({
+                assetData: createAssetData({
+                    bridgeAddress: randomAddress(),
+                }),
+            });
+            // This will actually revert when the AP tries to decode the return
+            // value.
+            return expect(tx).to.be.rejected();
+        });
+
+        it('fails if bridge reverts', async () => {
+            const revertError = 'FOOBAR';
+            const tx = transferFromAsync({
+                assetData: createAssetData({
+                    bridgeData: createBridgeData({
+                        revertError,
+                    }),
+                }),
+            });
+            return expect(tx).to.revertWith(revertError);
+        });
+
+        it('fails if balance of `to` increases by `amount - 1`', async () => {
```

Comment:
Nit: 
```suggestion
        it('fails if balance of `to` increases by less than `amount`', async () => {
```

For consistency with the above tests.